tests/Functional/GetRateTest.php

```diff
@@ -17,31 +17,30 @@
 
     public function testGetSpecificRateRecords()
     {
-        /** @var SpecificRateRecord $rates */
-        $rates = FixtureFactory::createPorter()->import(new ImportSpecification(new GetSpecificRate($this->apiKey)))
-            ->findFirstCollection();
+        $rate = FixtureFactory::createPorter()->importOne(new ImportSpecification(new GetSpecificRate($this->apiKey)));
 
-        $rateRecords = $rates->toAssociativeArray();
+        $this->assertCount(1, $rate);
 
-        $this->assertCount(1, $rateRecords);
-        $this->assertArrayHasKey('asset_id_base', $rateRecords[0]);
-        $this->assertArrayHasKey('asset_id_quote', $rateRecords[0]);
-        $this->assertArrayHasKey('rate', $rateRecords[0]);
-        $this->assertArrayHasKey('time', $rateRecords[0]);
+        $this->assertArrayHasKey('asset_id_base', $rate[0]);
+        $this->assertArrayHasKey('asset_id_quote', $rate[0]);
+        $this->assertArrayHasKey('rate', $rate[0]);
+        $this->assertArrayHasKey('time', $rate[0]);
     }
 
     public function testGetAllRateRecords()
     {
         /** @var AllRatesRecord $rates */
-        $rates = FixtureFactory::createPorter()->import(new ImportSpecification(new GetAllRate($this->apiKey)))
-            ->findFirstCollection();
+        $rates = FixtureFactory::createPorter()->import(new ImportSpecification(new GetAllRate($this->apiKey)));
 
-        $this->assertSame('BTC', $rates->getBase());
+        $this->assertCount(1267, $rates);
```

Comment:
This magic number is too brittle. The test needs to prove the count is correct without breaking whenever a rate is added or removed. You have two options, either:

1. Count the number of records during iteration and ensure it matches the reported count. To do this, the assertion would have to be moved after the loop.
2. Estimate the number of expected values, e.g. `self::assertGreaterThan(1200, count($rates))`.